record_batch.go

```diff
@@ -94,6 +94,7 @@ func (b *RecordBatch) encode(pe packetEncoder) error {
 
 	if err := pe.pop(); err != nil {
 		return err
+	} else {
```

Comment:
Was this empty `else` forgotten by any chance?

Reply:
ohh thanks, some left overs of debugging